bofire/surrogates/mlp_classifier.py

```diff
@@ -0,0 +1,220 @@
+from bofire.data_models.surrogates.api import MLPEnsemble as DataModel
+from bofire.surrogates.botorch import BotorchSurrogate
+from bofire.surrogates.single_task_gp import get_scaler
+from bofire.surrogates.trainable import TrainableSurrogate
+from bofire.utils.torch_tools import tkwargs
+
+
+class ClassificationDataSet(Dataset):
+    """
+    Prepare the dataset for classification
+    """
+
+    def __init__(self, X: Tensor, y: Tensor):
+        self.X = X.to(**tkwargs)
+        self.y = y.to(**tkwargs)
+
+    def __len__(self):
+        return len(self.X)
+
+    def __getitem__(self, i: int):
+        return self.X[i], self.y[i]
+
+
+class MLPClassifier(nn.Module):
+    def __init__(
+        self,
+        input_size: int,
+        output_size: int = 1,
+        hidden_layer_sizes: Sequence = (100,),
+        dropout: float = 0.0,
+        activation: Literal["relu", "logistic", "tanh"] = "relu",
+    ):
+        super().__init__()
+        if activation == "relu":
+            f_activation = nn.ReLU
+        elif activation == "logistic":
+            f_activation = nn.Sigmoid
+        elif activation == "tanh":
+            f_activation = nn.Tanh
+        else:
+            raise ValueError(f"Activation {activation} not known.")
+        layers = [
+            nn.Linear(input_size, hidden_layer_sizes[0]).to(**tkwargs),
+            f_activation(),
+        ]
+        if dropout > 0.0:
+            layers.append(nn.Dropout(dropout))
+        if len(hidden_layer_sizes) > 1:
+            for i in range(len(hidden_layer_sizes) - 1):
+                layers += [
+                    nn.Linear(hidden_layer_sizes[i], hidden_layer_sizes[i + 1]).to(
+                        **tkwargs
+                    ),
+                    f_activation(),
+                ]
+                if dropout > 0.0:
+                    layers.append(nn.Dropout(dropout))
+        layers.append(nn.Linear(hidden_layer_sizes[-1], output_size).to(**tkwargs))
+        self.layers = nn.Sequential(*layers)
+
+    def forward(self, x):
+        return nn.functional.log_softmax(self.layers(x), dim=1)
+
+
+class _MLPClassifierEnsemble(EnsembleModel):
+    def __init__(self, mlps: Sequence[MLPClassifier]):
+        super().__init__()
+        if len(mlps) == 0:
+            raise ValueError("List of mlps is empty.")
+        num_in_features = mlps[0].layers[0].in_features
+        num_out_features = mlps[0].layers[-1].out_features
+        for mlp in mlps:
+            assert mlp.layers[0].in_features == num_in_features
+            assert mlp.layers[-1].out_features == num_out_features
+        self.mlps = mlps
+        # put all models in eval mode
+        for mlp in self.mlps:
+            mlp.eval()
+
+    def forward(self, X: Tensor):
+        r"""Compute the model output at X.
+
+        Args:
+            X: A `batch_shape x n x d`-dim input tensor `X`.
+
+        Returns:
+            A `batch_shape x s x n x m x C`-dimensional output tensor where
```

Comment:
and what is n and m?

Reply:
Copy paste typo - I will fix these as it should just be `batch_size x n x C`